relay/cache/cache_accessor.go

```diff
@@ -55,17 +59,29 @@ type cacheAccessor[K comparable, V any] struct {
 
 	// accessor is the function used to fetch values that are not in the cache.
 	accessor Accessor[K, V]
+
+	// insertionTimes is a map of keys to the time they were inserted into the cache. Used to calculate the average
+	// lifespan of items in the cache.
+	insertionTimes map[K]time.Time
```

Comment:
Is it going to keep the insertion times for every single key?

Reply:
Just the keys currently in the cache. When the cache evicts something, it uses the time to determine how long the item was in the cache, and then it removes it from this map.